### ISSUE TITLE
Cannot enable rubocop-rspec derived cops on DisabledByDefault

### ISSUE THREAD

**Issue by r7kamura** [rubocop-hq/rubocop#9244]
To enable only certain cops derived from rubocop-rspec, I configured `DisabledByDefault: true` and tried to enable `RSpec/EmptyExampleGroup` like this:

```yaml
# .rubocop.yml
require:
  - rubocop-rspec

AllCops:
  DisabledByDefault: true

RSpec/EmptyExampleGroup:
  Enabled: true
```

ref: https://github.com/rubocop-hq/rubocop/pull/7923

## Expected behavior

I expected `RSpec/EmptyExampleGroup` to be enabled like this:

```console
$ bundle install
$ bundle exec rubocop
Inspecting 3 files
.C.

Offenses:

spec/example_spec.rb:2:3: C: RSpec/EmptyExampleGroup: Empty example group detected.
  context 'with empty example group' do
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

3 files inspected, 1 offense detected
```

## Actual behavior

But `RSpec/EmptyExampleGroup` is not enabled:

```console
$ bundle install
$ bundle exec rubocop
Inspecting 3 files
...

3 files inspected, no offenses detected
```

## Steps to reproduce the problem

```ruby
# Gemfile
source 'https://rubygems.org'

gem 'rspec'
gem 'rubocop'
gem 'rubocop-rspec'
```

```yaml
# .rubocop.yml
require:
  - rubocop-rspec

AllCops:
  DisabledByDefault: true

RSpec/EmptyExampleGroup:
  Enabled: true
```

```ruby
# spec/example_spec.rb
RSpec.describe 'Example' do
  context 'with empty example group' do
  end
end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.7.2 x86_64-linux)
  - rubocop-rspec 2.0.1
```

**Comment by r7kamura** [comment]
This problem doesn't happen on rubocop-rails. I think this is because rubocop-rspec has `RSpec` department in its config/default.yml, while rubocop-rails does not..

- https://github.com/rubocop-hq/rubocop-rspec/blob/56d801cffb74a9e89876336932e196eb128d9209/config/default.yml#L2
- https://github.com/rubocop-hq/rubocop-rails/blob/39bd3939a128c027d26dbfd95a98af19074a818f/config/default.yml

Because of this difference, RSpec `department['Enabled']` returns `false`, while Rails `department['Enabled']` returns `nil` .

https://github.com/rubocop-hq/rubocop/blob/f703e77ef32312c68e6fdcffcd95eb80e38200a2/lib/rubocop/config.rb#L283-L289

**Comment by dvandersluis** [comment]
This was really hairy but I have a PR incoming.
